### PR TITLE
48 implementation of a UI to display the friends list

### PR DESCRIPTION
**Pull request by Santboneless**

### CODE REVIEW

File: app/src/main/java/com/github/se/stepquest/screens/AddFriendScreenDummy.kt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                  horizontalArrangement = Arrangement.SpaceBetween,
+                  verticalAlignment = Alignment.CenterVertically) {
+                    Text(
+                        text = "Back",
+                        fontSize = 20.sp,
+                        modifier = Modifier.clickable { onDismiss() })
+                    Spacer(modifier = Modifier.weight(0.3f))
+                    IconButton(
+                        onClick = {
+                          onDismiss()
```
Comment (Liototo) [inline app/src/main/java/com/github/se/stepquest/screens/AddFriendScreenDummy.kt:57]: That line isn't necessary; from my tests just calling onSecondScreenDismiss() works to close both windows 

Also showAddFriendScreen is reset to false anyway when closing FriendsListDialog so the behaviour is consistent when clicking "Friends List" again (if that's why you did it)

File: app/src/main/java/com/github/se/stepquest/screens/FriendsListPopUp.kt [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+data class Friend(val name: String, val profilePictureUrl: String, val status: Boolean)
+
+@Composable
+fun FriendsListDialog(onDismiss: () -> Unit, friendsList: List<Friend>) {
+  val blueThemeColor = colorResource(id = R.color.blueTheme)
+  var showAddFriendScreen by remember { mutableStateOf(false) }
+  Dialog(onDismissRequest = { onDismiss() }) {
+    if (showAddFriendScreen) {
+      AddFriendScreen(
+          onDismiss = { showAddFriendScreen = false }, onSecondScreenDismiss = onDismiss)
```
Comment (Liototo) [inline app/src/main/java/com/github/se/stepquest/screens/FriendsListPopUp.kt:47]: Also maybe rename the different onDismiss functions so that what they do is a bit clearer because lines like this get a bit hard to follow imo

**Review by Liototo — COMMENTED** [review]
Other than the two nitpicks I have it looks pretty good, just don't forget to write tests as well!

**Review by Raytimz — APPROVED** [review]
Code looks clean. New features work on my end system.
